boa/src/syntax/parser/function/mod.rs

```diff
@@ -27,30 +27,35 @@ use crate::{
 };
 use std::io::Read;
 
-/// Formal parameters parsing.
+/// UniqueFormalParameters / FormalParameters parsing.
+///
+/// Due to the similarity between these productions we use a unique flag rather than
+/// duplicating the code.
```

Comment:
Actually, I would prefer having the former `FormalParameters`, and then having another `UniqueFormalParameters` structure that would basically call `FormalParameters::parse()`, and would then do a check on the generated parameter list before returning the result.